pytest_plugins/rerun_rp/rerun_rp.py

```diff
@@ -100,58 +58,108 @@ def pytest_addoption(parser):
     '''
     parser.addoption("--user", nargs='?', help=help_text)
     help_text = '''
-        Rerun Upgrade launch tests instead of Regular Tier launch from Report Portal,
+        Uses given launch as a reference for parsing RP tests to be used with
+        --only-failed, --only-skipped or --user
+
+        Usage: --rp-reference-launch-uuid [value]
+
+        Value: report_portal_launch_uuid
+    '''
+    parser.addoption("--rp-reference-launch-uuid", nargs='?', help=help_text)
+    help_text = '''
+        Uses given satellite version + snap as a reference for parsing RP tests to be used with
+        --only-failed, --only-skipped or --user
+        e.g 6.10.0-1.0
 
-        Usage: --upgrades-rerun
+        Usage: --sat-version [value]
+
+        Value: sat_version
     '''
-    parser.addoption("--upgrades-rerun", action='store_true', default=False, help=help_text)
+    parser.addoption("--sat-version", help=help_text)
 
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_collection_modifyitems(items, config):
     """
-    Collects and modifies tests collection based on pytest options to select tests marked as
-    failed/skipped and user specific tests in Report Portal
+    Collects and modifies test collection based on the pytest options to select the tests marked as
+    failed/skipped and user-specific tests in Report Portal
     """
+    rp_url = settings.report_portal.portal_url or config.getini('rp_endpoint')
+    rp_uuid = getenv('RP_UUID') or config.getini('rp_uuid') or settings.report_portal.api_key
+    rp_launch_name = (
+        getenv('RP_LAUNCH') or settings.report_portal.launch_name or config.getini('rp_launch')
+    )
+    rp_project = config.getini('rp_project') or settings.report_portal.project
     fail_args = config.getoption('only_failed', False)
     skip_arg = config.getoption('only_skipped', False)
     user_arg = config.getoption('user', False)
-    upgrades_rerun = config.getoption('upgrades_rerun', False)
-    if not any([fail_args, skip_arg, user_arg, upgrades_rerun]):
```

Comment:
How are we handling the upgrades rerun ? 

A help message can be updated on how to rerun the upgrade launch tests though.

Reply:
How would upgrade be different here?